lib/ace/transport_app.rb

```diff
@@ -33,6 +34,58 @@ def initialize(config = nil)
       super(nil)
     end
 
+    # Initialises the puppet target.
+    # @param certname   The certificate name of the target.
+    # @param transport  The transport provider of the target.
+    # @param target     Target connection hash or legacy connection URI
+    # @return [Puppet device instance] Returns Puppet device instance, or nil if exception thrown
+    # @raise  [puppetlabs/ace/invalid_param] If nil parameter or no connection detail found
+    # @example Connect to device.
+    #   init_puppet_target('test_device.domain.com', 'panos',  JSON.parse("target":{
+    #                                                                       "remote-transport":"panos",
+    #                                                                       "host":"fw.example.net",
+    #                                                                       "user":"foo",
+    #                                                                       "password":"wibble"
+    #                                                                     }) ) => panos.device
+    def self.init_puppet_target(certname, transport, target)
+      unless target
+        ACE::Error.new("There was an error parsing the Puppet target. 'target' not found",
```

Comment:
These errors are not being raised. You must call `raise ACE::Error.new(.)` then the nil return will not be required.

Reply:
Thanks David, now raising